packages/components/src/entities/SampleFinderSection.tsx

```diff
@@ -508,52 +506,20 @@ interface SampleFinderSamplesProps extends SampleFinderSamplesGridProps {
 }
 
 const SampleFinderSamplesImpl: FC<SampleFinderSamplesGridProps & InjectedQueryModels> = memo(props => {
-    const { actions, columnDisplayNames, queryModels, gridButtons, gridButtonProps } = props;
-    const [isLoading, setIsLoading] = useState<boolean>(true);
-
+    const { actions, queryModels, gridButtons, gridButtonProps } = props;
+    const [tabRowCounts, setTabRowCounts] = useState<{ [key: string]: number }>(undefined);
     useEffect(() => {
-        const allModelsLoaded = Object.values(queryModels).filter(model => model.isLoading).length == 0;
-        if (allModelsLoaded && isLoading) {
-            const promises = [];
-            Object.values(queryModels).forEach(queryModel => {
-                const { hasUpdates, columns } = getFinderViewColumnsConfig(queryModel, columnDisplayNames);
-                if (hasUpdates) {
-                    promises.push(saveFinderGridView(queryModel.schemaQuery, columns));
-                }
-            });
-            try {
-                Promise.all(promises)
-                    .then(schemaQueries => {
-                        // since we have updated views, we need to invalidate the details cache so we pick up the new views
-                        schemaQueries.forEach(schemaQuery => {
-                            invalidateQueryDetailsCache(schemaQuery);
-                        });
-                        setIsLoading(false);
-                    })
-                    .catch(reason => {
-                        console.error('Error saving all finder views.', reason);
-                        setIsLoading(false);
-                    });
-            } catch (error) {
-                // ignore: already logged
-            }
-        }
-    }, [queryModels]);
+        if (Object.keys(queryModels).length <= 2)
+            // if 1 All samples + 1 type, rely on autoload count
+            return;
 
-    useEffect(() => {
-        return () => {
-            if (queryModels) {
-                for (const queryModel of Object.values(queryModels)) {
-                    (async () => {
-                        try {
-                            await removeFinderGridView(queryModel);
-                        } catch (error) {
-                            // ignore; already logged
-                        }
-                    })();
-                }
-            }
-        };
+        getSampleFinderTabRowCounts(queryModels)
+            .then(counts => {
+                setTabRowCounts(counts);
+            })
+            .catch(error => {
+                console.error(error);
```

Comment:
`getSampleFinderTabRowCounts(queryModels)` already console logs the error so no need to log again here. General rule of thumb is to only log at the lowest level.